server/user.go

```diff
@@ -299,36 +318,48 @@ func (s *UserMgmtServer) getCreds(r *http.Request, requiresAdmin bool) (api.Cred
 		return api.Creds{}, api.ErrorUnauthorized
 	}
 
-	usr, err := s.um.Get(sub)
-	if err != nil {
-		if err == user.ErrorNotFound {
-			return api.Creds{}, api.ErrorUnauthorized
-		}
-		log.Errorf("userMgmtServer: GetCreds err: %q", err)
-		return api.Creds{}, err
-	}
-
-	i := 0
-	for _, clientID := range clientIDs {
-		// Make sure the client actually exists.
-		isAdmin, err := s.cm.IsDexAdmin(clientID)
+	var usr user.User
+	if s.allowClientCredsAuth && (len(clientIDs) == 1) && (sub == clientIDs[0]) {
```

Comment:
Wait, why are you extracting this information from a bearer token? I would think that this endpoint would use one of the many client authentication methods provided by OpenID Connect. This just looks like it should be any random ID Token issued to the client.

https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication